### DISCUSSION TITLE
Mocking Star: Empower Your Testing with Automated Network Mocking

### DISCUSSION THREAD

**Discussion by yusufozgul** [MobileNativeFoundation/discussions#211] [General]
Two years ago, when we started writing UI tests, we attempted to solve the mock data problem using the https://github.com/httpswift/swifter library. However, as time went on, we decided that Swifter was not suitable for our needs. The main reasons for this were as follows:

- Difficulty in providing different responses for requests with the same URL.
- Providing mocks based on path parameters.
- When UI tests initially started running, it took around 10 seconds for Swifter to start responding.
- The need to manually create mock data and provide it to Swifter.
- The need to start the entire process from scratch to update saved mocks.

For these reasons, we decided to develop our own application. At this point, Trendyol iOS application has:
- **~610 Smoke Tests**
- **~1225 Snapshot Tests**

All tests perform approximately **130k executions** weekly on average. This means Mocking Star handles millions of requests weekly and ensures all our UI tests run smoothly. Mocking Star has solved all our mock data problems and greatly facilitated the maintenance of UI tests.

We have open-sourced the Mocking Star application. Here are some important features:

- Easy integration with iOS and Android applications.
- User-friendly macOS application.
- CLI version for CI/CD environments.
- Plugin system that you can customize according to your needs.

### Links:
- [Mocking Star](https://github.com/Trendyol/mockingstar)
- [Docs](https://trendyol.github.io/mockingstar/documentation/mockingstar/documentation)
- [iOS Client Library](https://github.com/Trendyol/mockingstar-ios)
- [Android Client Library](https://github.com/Trendyol/mockingstar-android)

**Comment by natnaelbelayneh** [comment]
right
